app/helpers/sessions_helper.rb

```diff
@@ -3,17 +3,36 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
     if session[:user_id]
-      @current_user ||= User.find_by(id: session[:user_id])
+      @current_user ||= User.find_by id: session[:user_id] if session[:user_id]
```

Comment:
```suggestion
      @current_user ||= User.find_by id: session[:user_id]
```
remove it because u use `if session[:user_id]` on above line

Reply:
understood. thank you